src/Writer/Stream.php

```diff
@@ -76,7 +76,7 @@ public function __construct($streamOrUrl, $mode = null, $logSeparator = null, $f
             }
 
             $this->stream = $streamOrUrl;
-        } else {
+        } elseif (is_string($streamOrUrl)) {
```

Comment:
is this related to PHP 8 or just a random bugfix?
If the latter, please create a dedicated PR regarding this bug.

Reply:
its related to PHP 8 because in the testcase `xml_parser_create` is used. 
The function will return in PHP8 an XmlParser object instead of a resource. See https://3v4l.org/tc867 

To prevent type errors i added the `is_string` check. If the input is an Object you will get the `\Laminas\Log\Exception\InvalidArgumentException` instead of an typerror.
> Fatal error: Uncaught TypeError: fopen(): Argument 1 ($filename) must be of type string, XmlParser given

Any other ideas to handle this case? I think the library InvalidArgumentException is better than a TypeError?